vgscheckout/src/main/java/com/verygoodsecurity/vgscheckout/networking/command/core/CompositeCommand.kt

```diff
@@ -0,0 +1,53 @@
+package com.verygoodsecurity.vgscheckout.networking.command.core
+
+import android.content.Context
+import com.verygoodsecurity.vgscheckout.exception.VGSCheckoutException
+import com.verygoodsecurity.vgscheckout.networking.command.GetOrderDetails
+import com.verygoodsecurity.vgscheckout.networking.command.GetSavedCardsCommand
+
+internal class CompositeCommand(
+    context: Context,
+    private val commands: Collection<Command<*, *>>
+) : Command<CompositeCommand.Params, CompositeCommand.Result>(context) {
+
+    private var processedCommandsCounter = 0
+
+    fun execute(onResult: (Result) -> Unit) {
+        execute(Params(commands), onResult)
+    }
+
+    override fun run(params: Params, onResult: (Result) -> Unit) {
+        params.commands.forEach { command ->
+            when (command) {
+                is GetSavedCardsCommand -> command.execute {
```

Comment:
@Endorf I think in this case, we can create not`CompositeCommand`, but for example `GetPaymentDetailsCommand` etc., and make two requests and only then return a result. I think  `CompositeCommand` name is too general for this case.